File: server/src/routers/color.ts
import { ColorRoute } from '@/enums';
import { useColor } from '@/services';
import { Hono } from 'hono';

const service = useColor();
const color = new Hono();

color.get(ColorRoute.RETRIEVE_COLORS, async (c) => c.json(await service.list()));

export { color };

File: server/internal/controllers/color.go
package controllers

import (
	"encoding/json"
	"net/http"
	"sukejuuru/internal/services"
)

type colorController struct{}

func UseColor() colorController {
	return colorController{}
}

func (ctrl colorController) RetrieveColors(w http.ResponseWriter, r *http.Request) {
	result, err := services.UseColor().FindAll()
	if err != nil {
		http.Error(w, err.Error(), http.StatusInternalServerError)
		return
	}
	response, err := json.Marshal(result)
	if err != nil {
		http.Error(w, err.Error(), http.StatusInternalServerError)
		return
	}
	w.WriteHeader(http.StatusOK)
	w.Write([]byte(response))
}
